src/main/java/mate/academy/dao/impl/CinemaHallDaoImpl.java

```diff
@@ -0,0 +1,55 @@
+package mate.academy.dao.impl;
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.CinemaHall;
+import mate.academy.util.HibernateUtil;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class CinemaHallDaoImpl implements CinemaHallDao {
+    @Override
```

Comment:
Maybe it's better to use SessionFactory as singleton.
```suggestion
     private static SessionFactory sessionFactory = HibernateUtil.getSessionFactory().openSession();

    @Override
```

Reply:
fixed